azure/converters/vmss.go

```diff
@@ -34,20 +35,20 @@ const (
 )
 
 // SDKToVMSS converts an Azure SDK VirtualMachineScaleSet to the AzureMachinePool type.
-func SDKToVMSS(sdkvmss compute.VirtualMachineScaleSet, sdkinstances []compute.VirtualMachineScaleSetVM) azure.VMSS {
+func SDKToVMSS(sdkvmss armcompute.VirtualMachineScaleSet, sdkinstances []compute.VirtualMachineScaleSetVM) azure.VMSS {
```

Comment:
This is weird: mixing SDKv2 and SDKv1 parameters. But it seems to work and allows us to keep the whole virtualmachines service conversion separate, for a future PR.